magda-gateway/src/Authenticator.ts

```diff
@@ -185,6 +186,65 @@ export default class Authenticator {
         res.clearCookie(DEFAULT_SESSION_COOKIE_NAME, deleteCookieOptions);
     }
 
+    /**
+     * A middleware that:
+     * - validate the session and destory invalid session
+     * - recover session data for valid session
+     *
+     * @param {express.Request} req
+     * @param {express.Response} res
+     * @param {express.NextFunction} next
+     * @returns
+     * @memberof Authenticator
+     */
+    validAndRecoverSession(
```

Comment:
Yeah~ I just quickly put one name --- I think we probably need a better name.
Do you have a better one (Otherwise I will just change it to `validateAndRecoverSession`)?
Basically, it does the followings:
- Validate the session and destroy the invalid one (so the future request won't carry cookies)
- If it's valid session, handle over to `this.passportMiddleware` and `this.passportSessionMiddleware` to build up full session env (i.e. pull data and set them probably to `req.user`)

Reply:
ah right. I guess personally I'd call that `validateAndRefreshSession`? Naming things is hard  :(

Reply:
Sounds good, updated 👍